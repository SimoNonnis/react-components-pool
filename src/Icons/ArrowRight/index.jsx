import React from 'react';
import PropTypes from 'prop-types';

/**
 * ArrowRight Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function ArrowRight({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 20 20"
      width={width}
      height={height}
      className={className}
    >
      <title>ArrowRight Icon</title>

      <path
        className={classNamePath}
        fill={fill}
        d="M5.48,18.48a.5.5,0,0,1-.33-.88L13.77,10,5.15,2.4a.5.5,0,1,1,.66-.75l9,8a.5.5,0,0,1,0,.75l-9,8A.5.5,0,0,1,5.48,18.48Z"
      />
    </svg>
  );
}

ArrowRight.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

ArrowRight.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'arrowRightIcon',
  classNamePath: 'arrowRightIconPath',
};

export default ArrowRight;
