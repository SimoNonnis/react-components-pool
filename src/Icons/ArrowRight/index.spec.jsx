/* eslint-env jest */
import React from 'react';

import ArrowRight from './';

describe('Test <ArrowRight /> icon', () => {
  it('should render ArrowRight icon correctly with default props', () => {
    const wrapper = shallow(<ArrowRight />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ArrowRight icon correctly with props', () => {
    const wrapper = shallow(
      <ArrowRight fill="#a4a1a1" className="testClassname" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
