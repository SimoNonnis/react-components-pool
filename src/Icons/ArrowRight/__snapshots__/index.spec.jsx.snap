// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ArrowRight /> icon should render ArrowRight icon correctly with default props 1`] = `
<svg
  className="arrowRightIcon"
  height={40}
  viewBox="0 0 20 20"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    ArrowRight Icon
  </title>
  <path
    className="arrowRightIconPath"
    d="M5.48,18.48a.5.5,0,0,1-.33-.88L13.77,10,5.15,2.4a.5.5,0,1,1,.66-.75l9,8a.5.5,0,0,1,0,.75l-9,8A.5.5,0,0,1,5.48,18.48Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <ArrowRight /> icon should render ArrowRight icon correctly with props 1`] = `
<svg
  className="testClassname"
  height={30}
  viewBox="0 0 20 20"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    ArrowRight Icon
  </title>
  <path
    className="arrowRightIconPath"
    d="M5.48,18.48a.5.5,0,0,1-.33-.88L13.77,10,5.15,2.4a.5.5,0,1,1,.66-.75l9,8a.5.5,0,0,1,0,.75l-9,8A.5.5,0,0,1,5.48,18.48Z"
    fill="#a4a1a1"
  />
</svg>
`;
